File: algorithm/LeetCode/CountPrimes.cpp
class Solution {
public:
    int countPrimes(int n) {
        vector<int> vtor(n + 1, 0);
        vector<int> ret;
        for (int i = 0; i <= n; ++i)
            vtor[i] = i;
        for (int i = 2; i < n; ++i){//边界条件注意
            if (vtor[i] != -1){
                ret.push_back(vtor[i]);
                int tmpPrime = vtor[i];
                for (int mul = 1; tmpPrime * mul <= n; mul++){
                    vtor[tmpPrime * mul] = -1;
                }
            }
        }
        return ret.size();
    }
};

File: algorithm/LeetCode/CountPrimes/CountPrimes.java
public class Solution {
    public int countPrimes(int n) {
        int [] prime = new int[n+1]; //这里其实用boolean更好，懒得改了
        int count = 0;
        for(int i = 0; i < n+1; ++i){
            prime[i] = i;
        }
        for(int i = 2; i < n+1; ++i){
            if(prime[i] == -1)
                continue;
            else{
                count++;
                for(int mul = 2; mul * i < n+1; ++mul){
                    prime[mul*i] = -1;
                }
            }
        }
        return count;
    }
}